tcms/testcases/tests/test_models.py

```diff
@@ -157,3 +159,28 @@ def test_send_mail_to_case_author(self, send_mail):
             recipients,
             fail_silently=False,
         )
+
+
+class TestActualDurationProperty(TestCase):
+    """Test TestExecution.actual_duration"""
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.test_execution = TestExecution()
+        cls.test_execution.start_date = timezone.now()
+        cls.test_execution.stop_date = timezone.now() + timezone.timedelta(days=1, seconds=0, microseconds=0)
+
+    def test_calculation_of_actual_duration(self):
+        self.assertEqual(
+            self.test_execution.actual_duration, timezone.timedelta(days=1, seconds=0, microseconds=2)
+        )
+
+    def test_actual_duration_empty_start_date(self):
+        empty_start_date_actual_duration = TestExecution(start_date=None)
+        self.assertEqual(empty_start_date_actual_duration.actual_duration, None)
+
+    def test_actual_duration_empty_stop_date(self):
+        empty_stop_date_actual_duration = TestExecution(stop_date=None)
+        self.assertEqual( empty_stop_date_actual_duration.actual_duration, None)
```

Comment:
Last 2 scenarios look OK. The tests are failing because the implementation doesn't take into account that some/all of the values can be None.  Now you have to update the implementation of this property to account for edge cases which should make these 2 test scenarios pass.